### PR TITLE
refactor: util module

### PR DESCRIPTION
**Pull request by nabetti1720**
### Description of changes

- The declaration and implementation (`TextEncoder` / `TextDecoder`) parts of the `util` module included in `llrt_core` have been integrated and reorganized.
- As for the migration to `llrt_modules`, it is difficult at this time due to blocking by `format_plain()` included in the `console` module (code for Lambda and other code calls for common processing, which requires consideration). I am going to pass this up for this time as I would like to make it my next effort.

### Checklist

- [x] Created unit tests in `tests/unit` and/or in Rust for my feature if needed
- [x] Ran `make fix` to format JS and apply Clippy auto fixes
- [x] Made sure my code didn't add any additional warnings: `make check`
- [x] Added relevant type info in `types/` directory
- [x] Updated documentation if needed ([API.md](API.md)/[README.md](README.md)/Other)

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Comment by Sytten** [comment]
I didnt migrate it specifically because console is tied to AWS stuff.
The way to handle this is via settings using use data tied to the runtime. This was added to rquickjs recently but I didnt have time to finish my new module trait.
I would hold off on migrating this module just yet.

**Comment by nabetti1720** [comment]
> I didnt migrate it specifically because console is tied to AWS stuff. The way to handle this is via settings using user data tied to the runtime. This was added to rquickjs recently but I didnt have time to finish my new module trait. I would hold off on migrating this module just yet.

Sorry if I have misled you. The goal of this PR was to get this far from the beginning. When I said I wanted to work on the console module “next,” I meant the next PR, not this one. However, your comments suggest that this would be a better solution than I have in mind, using the functionality implemented in rquickjs.

This PR is only a minor change, both as a feature available to users and as source code to be managed under the Util module, but I'll leave it up to you guys to decide if you want to merge it or not. :)

**Review by richarddavison — APPROVED** [review]
LGTM

**Pull request merged**
